### PR TITLE
[#562][DotNet] Add Azure Storage Blobs tests to the test solution

### PR DESCRIPTION
**Pull request by sw-joelmut**
Addresses # 562

## Description
This PR adds the `Azure Storage Blobs` tests to the new `Integration` test project.

### Detailed Changes
- Added `Microsoft.Bot.Builder.Azure.Blobs` dependency to the `Integration.csproj`.
- Moved `CosmosDbBaseTests` and `CosmosDbStorageItem` to the Storage folder, allowing reusing all the tests related to classes that implements `IStorage`.
  - The base tests have been renamed and marked as `Fact`, so, when other classes inherit from it, it will inject automatically all the tests for a specific `Storage`.
  - Added support for testing multiple `Storage` cases (e.g default config or with _TypeNameHandlingNone_ in the _JsonSerializer_).
- Updated all `CosmosDb` related classes to support the new test structure change, multiple `Storage` configurations, and new added tests.
- Added new `Blobs Storage` and `Azure Blob Storage` tests with similar tests and fixture configuration as for `CosmosDb`. Also, detects if the Blobs service is up and running.

## Testing
The following images show the new Storage Blobs tests added to the Integration project, alongside the folder structure. 
image

### CODE REVIEW

File: Tests/Integration/DotNet/Azure/Storage/Blobs/BlobsStorageBaseFixture.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
```
Comment (ceciliaavila) [inline Tests/Integration/DotNet/Azure/Storage/Blobs/BlobsStorageBaseFixture.cs]: `System` and `Microsoft.Bot.Builder` are not being used.

**Review by ceciliaavila — CHANGES_REQUESTED** [review]
Just a detail, the rest looks great!

File: Tests/Integration/DotNet/Azure/Storage/Blobs/AzureBlobStorageTests.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Xunit;
+
+namespace IntegrationTests.Azure.Storage.Blobs
+{
+    [Trait("TestCategory", "Deprecated")]
```
Comment (JuanAr) [inline Tests/Integration/DotNet/Azure/Storage/Blobs/AzureBlobStorageTests.cs:8]: I'm not sure about this one. but the "Deprecated" trait ringed some bells.
If there's a good reason to keep this, ignore my comment. Just double checking.

**Review by JuanAr — APPROVED** [review]
Left a comment to double check something. But everything looks great.

File: Tests/Integration/DotNet/Azure/Storage/Blobs/AzureBlobStorageTests.cs (hunk 2 above)
Comment (sw-joelmut) [reply Tests/Integration/DotNet/Azure/Storage/Blobs/AzureBlobStorageTests.cs:8]: We are defining all the classes that are marked as deprecated in the sdk here as well in the tests, so it is easier to identify, follow, filter for v5, etc in the future.

File: Tests/Integration/DotNet/Azure/Storage/Blobs/BlobsStorageBaseFixture.cs (hunk 1 above)
Comment (sw-joelmut) [reply Tests/Integration/DotNet/Azure/Storage/Blobs/BlobsStorageBaseFixture.cs]: Applied, thanks @ceciliaavila

**Review by ceciliaavila — APPROVED** [review]

**Comment by sw-joelmut** [comment]
Closed in favor of #144

**Pull request closed without merging**